src/sagas/featureToggle.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Platform } from 'react-native';
+import { UnleashClient, EVENTS as UnleashEvents } from 'unleash-proxy-client';
+import { get } from 'lodash';
+
+import {
+  takeEvery,
+  all,
+  call,
+  delay,
+  put,
+  cancelled,
+  select,
+  race,
+  take,
+  fork,
+  spawn,
+} from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { getUniqueId } from 'react-native-device-info';
+import {
+  setUnleashClient,
+  setFeatureToggles,
+  featureToggleInitialized,
+} from '../actions';
+import {
+  UNLEASH_URL,
+  UNLEASH_CLIENT_KEY,
+  UNLEASH_POLLING_INTERVAL,
+  STAGE,
+  FEATURE_TOGGLE_DEFAULTS,
+} from '../constants';
+
+const CONNECT_TIMEOUT = 10000;
+const MAX_RETRIES = 5;
+
+export function* handleInitFailed(currentRetry) {
+  if (currentRetry >= MAX_RETRIES) {
+    console.error('Max retries reached while trying to create the unleash-proxy client.');
+    const unleashClient = yield select((state) => state.unleashClient);
+
+    if (unleashClient) {
+      unleashClient.close();
+      yield put(setUnleashClient(null));
+    }
+
+    // Even if unleash failed, we should allow the app to continue as it
+    // has defaults set for all feature toggles. Emit featureToggleInitialized
+    // so sagas waiting for it will resume.
+    yield put(featureToggleInitialized());
+    return;
+  }
+
+  yield spawn(monitorFeatureFlags, currentRetry + 1);
+}
+
+export function* fetchTogglesRoutine() {
+  while (true) {
+    // Wait first so we don't double-check on initialization
+    yield delay(UNLEASH_POLLING_INTERVAL);
+
+    const unleashClient = yield select((state) => state.unleashClient);
+
+    try {
+      yield call(() => unleashClient.fetchToggles());
+    } catch (e) {
+      // No need to do anything here as it will try again automatically in
```

Comment:
Why will it try again? Shouldn't you throw the error, so the `monitorFeatureFlags` catch captures it and retries?

Reply:
No, at this point, unleash is already initialized

What I mean is that we are not breaking the loop here, so it will try again in `UNLEASH_POLLING_INTERVAL`s

Reply:
It does, but it also provides a mechanism to disable it

I don't like the idea of a library making http requests behind the scenes with a `setInterval`

Reply:
Mmm... now I see it. Yes, it increases our control. Ok.